start: new experiments introduction (#2574)
iterative/dvc.org: content/docs/start/experiments-trail/experiments.md

---
title: 'Get Started: Experiments'
---

# Get Started with Experiments

> In machine learning projects, number of <abbr>experiments</abbr> grow rapidly.

Reviewer: ```suggestion
In machine learning projects, the number of <abbr>experiments</abbr> grows rapidly.
```

Author: Thank you :)

> DVC can track these experiments, list their most relevant parameters and

Reviewer 2: it helps to run them also, navigate, compare ...

Author: Added. Thanks.

metrics, and commit only the ones that we need to Git.

In this section, we explore the basic features of DVC experiment management with
[`get-started-experiments`][gse] project.

[gse]: https://github.com/iterative/get-started-experiments

<details>

> ## ⚙️ Installing the Example Project

Reviewer 2: should be `###` for the title and inside those sections I think

These commands are run in the [`get-started-experiments`][gse] project. You can
run the commands in this document after cloning the repository and installing
the requirements.

### ⚡ Clone the project and create virtual environment

Please clone the project and create a virtual environment.

> We create a virtual environment to keep the libraries we use isolated from the
> rest of your system. This prevents version conflicts.

```dvc

> $ git clone https://github.com/iterative/get-started-experiments -b get-started

Reviewer 2: why do we need `-b`?

Reviewer 3: idk - branch `get-started` doesn't even exist at https://github.com/iterative/example-dvc-experiments/branches

Author: It's a tag. https://github.com/iterative/example-dvc-experiments/tree/get-started

It shows the point to start this tutorial. I'm moving the tag to a particular commit instead of updating the name here.

Reviewer 2: okay, question - since we are not creating this repo from the very beginning like we do in the example get started one - does it make sense to let them just clone the latest version (and may be briefly explain core pieces of it)?

Author: We should start somewhere in the middle, before experiments. Otherwise, the experiments repo won't contain any experiments. We are also using this repository for `dvc exp pull/push` and I believe it's better to keep this tag and start from it. The alternative, that is starting from a repo with experiments in it leads to a more complicated narrative.

Reviewer 2: okay, makes sense ... I wonder if long term we can make this tag `get-started` the default state though fo the repo (and add experiments as "you go" in the docs)

$ cd get-started-experiments

> $ virtualenv .venv

Reviewer 2: it won't work on Mac (need to check Windows) - you need to provide Python 3

I would simplify this with `We strongly recommend creating a virtual environment first.` - the way we handle this in the current example get started

Author: `virtualenv` requires a separate installation in all systems. The standard one in Py3 is `venv`. We discussed this in https://github.com/iterative/example-repos-dev/issues/26 and decided to continue with `virtualenv`. 

I can also strongly recommend, if that's the point :)

Author: Even `venv` requires a separate `apt install python3-venv` in Debian and derivatives.

Reviewer 2: My question was about not providing so much details. We can keep `git clone` and maybe `pip install -r requirements` ... we can put a link nearby to the virtualenv like we do in the current get started?

$ . .venv/bin/activate
$ python -m pip install -r requirements.txt
```

### ✅ Get the data set

> The repository we cloned doesn't contain the dataset. We use `dvc pull` to

Reviewer 2: here we need a reference to the data management

Author: I believe we'll revise all the docs when we'll have done other trails. All steps will link to other relevant parts.

Reviewer 2: can we include it now? not even the link itself, but some phrase - that `dvc pull` is one of those commands to save, load, manage datasets and ML models ... read here ....

update the missing data files.

```dvc
$ dvc pull
```

The repository already contains the necessary configuration to run the
experiments.

> ## 💡 Preparing a project for DVC experiments

Reviewer: It's not immediately clear to me that these steps are already complete for the cloned repo. I guess this is more of a reference so that users know how you got to this point? I think it's important to clarify, or my first step would be `dvc init`, which will throw an error.

> If DVC is not installed to the system, please refer to [install](/doc/install).

Reviewer 2: we already mentioned `dvc pull` before?

Author: removed the section.

In this getting started page, we assume that there is already a configured DVC
project to simplify the introduction. DVC experiments require a DVC pipeline
defined in the project.

If DVC is not initialized before in the project, you can do so by:

```dvc
$ dvc init
```

DVC also requires commands to be run and their dependencies to be defined as
stages. We use `dvc stage add` to add a stage and set its dependencies.

```dvc
$ dvc stage add -n train \
                -p model.conv_units \
                -p train.epochs \
                -d data/images \
                -m metrics.json \

>                 python3 src/train.py

Reviewer 4: We should encourage users to update `dvc.yaml` instead of `stage add`. The command may seem useful for simple cases, but it may prevent users from learning our DSL which is what is driving all these pipelines interaction.

Related: #2578

```yaml
train:
  cmd: python3 src/train.py
  deps:
  - data/images
  params:
  - model.conv_units
  - train.epochs
  metrics:
  - metrics.json

Reviewer: I think the goal is to revise once we implement https://github.com/iterative/dvc/issues/6446 to make this simpler. @skshetry we should also consider as part of `dvc exp init` and possible as a broader topic how to provide templates or other starting points for `dvc.yaml`.

Reviewer 4: hmm, I am not sure if the `dvc exp init` is going to be any easier than the `dvc.yaml` for other scenarios. We should not try to keep them away from dvc.yaml.

Author: Editing `dvc.yaml` requires a node in the narrative towards pipelines, which we may not return to experiments easily. We are trying to come up with the simplest. The pipelines will have its own tutorial once we complete these. @skshetry

Reviewer 2: I think since we don't include pipelines, and we could start this doc like:

`git clone project`
`pip install req`
`dvc pull`
`dvc exp run`

do we need to explain anything at all here?

or at least if we want to keep doin this, I would instead create an expandable section:

"Explaining structure of the project" where we explain `dvc.yaml`, put a reference to the pipelines/dvc stage add, etc.

## 👟 Running the experiment with default parameters

The purpose of `dvc exp` subcommands is to run the experiments without
committing parameter and dependency changes to Git. The artifacts like models,
metrics produced by each experiment are tracked by DVC and persisted on demand.

Running the experiment with the default project settings requires only the
command:

```dvc
$ dvc exp run
...
Reproduced experiment(s): exp-b28f0
Experiment results have been applied to your workspace.
...
```

It runs the specified command (`python train.py`) in `dvc.yaml`. That command
writes the metrics values to `metrics.json`.

This experiment is then associated with the values found in parameters file
(`params.yaml`), and other dependencies (`data/images/`) with these produced
metrics.

<details>

### 📖 If you used `dvc repro` before

Earlier versions of DVC uses `dvc repro` to run the pipeline. If you already
have a DVC project, you may already be using `dvc repro`.

In DVC 2.0 `dvc exp run` supersedes `dvc repro`. We use `dvc repro` to run the
pipeline as found in the <abbr>workspace</abbr>. All the parameters and
dependencies are retrieved from the current workspace. It doesn't use any

> special objects to track the experiments or associate parameters with metrics.

Reviewer: I don't fully understand the 2nd-4th sentences of this paragraph and which ones refer to `repro` vs `exp run` behavior. I'm not sure they are needed, since the 5th sentence onwards makes clear to me the rationale for using `exp run`.

Author: Removed the first sentence that may be causing the confusion.

batch. This is especially handy when you have long running experiments.

We add experiments to the queue using the `--queue` option of `dvc exp run`. We
also use `-S` (`--set-param`) to set a value for the parameter.

```dvc
$ dvc exp run --queue -S model.conv_units=32
Queued experiment '3cac8c6' for future execution.
$ dvc exp run --queue -S model.conv_units=64
Queued experiment '23660b6' for future execution.
$ dvc exp run --queue -S model.conv_units=128
Queued experiment '6591a57' for future execution.
$ dvc exp run --queue -S model.conv_units=256
Queued experiment '9109ea9' for future execution.
```

Next, run all (`--run-all`) queued experiments in parallel. You can specify the
number of parallel processes using `--jobs`:

```dvc
$ dvc exp run --run-all --jobs 2
```

</details>

## ↔️ Comparing experiments

The experiments are run several times with different parameters. We use
`dvc exp show` to compare all of these experiments. This command presents the
parameters and metrics produced in experiments in a nicely formatted table.

```dvc
$ dvc exp show
```

```dvctable
┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓

> ┃ white:**Experiment**    ┃ white:**Created**┃    yellow:**loss** ┃    yellow:**acc** ┃ blue:**train.epochs** ┃ blue:**model.conv_units** ┃

Reviewer 5: Looks like certain parts of the table are missing some white space, causing the table to not align correctly. 

```suggestion
┃ white:**Experiment**              ┃ white:**Created**      ┃ yellow:**loss**    ┃ yellow:**acc**    ┃ blue:**train.epochs** ┃ blue:**model.conv_units** ┃
```

┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
│ workspace               │ -            │ 0.23508 │ 0.9151 │ 10           │ 24               │
│ 7317bc6                 │ Jul 18, 2021 │       - │      - │ 10           │ 16               │
│ ├── e2647ef [exp-ee8a4] │ 05:14 PM     │ 0.23146 │ 0.9145 │ 10           │ 64               │
│ ├── 15c9451 [exp-a9be6] │ 05:14 PM     │ 0.25231 │ 0.9102 │ 10           │ 32               │
│ ├── 9c32227 [exp-17dd9] │ 04:46 PM     │ 0.23687 │ 0.9167 │ 10           │ 256              │
│ ├── 8a9cb15 [exp-29d93] │ 04:46 PM     │ 0.24459 │ 0.9134 │ 10           │ 128              │
│ ├── dfc536f [exp-a1bd9] │ 03:35 PM     │ 0.23508 │ 0.9151 │ 10           │ 24               │
│ └── 1a1d858 [exp-6dccf] │ 03:21 PM     │ 0.23282 │ 0.9152 │ 10           │ 16               │
└─────────────────────────┴──────────────┴─────────┴────────┴──────────────┴──────────────────┘
```

> By default it shows all the parameters and the metrics with the timestamp. If

Reviewer: Could this go inside a `<details>` tag? Feel free to ignore if you think it's better in the main text.

Author: Need a new header to this paragraph to put into a `<details>` section. Let me look at it once more.

you have large number of parameters, metrics or experiments, this may lead to a
cluttered view. You can limit the table to specific metrics, or parameters, or
hide the timestamp column with `--include-metrics`, `--include-params`, or
`--no-timestamp` options of the command, respectively.

```dvc
$ dvc exp show --no-timestamp \
  --include-params model.conv_units --include-metrics acc
```

```dvctable
┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
┃ white:**Experiment**              ┃    yellow:**acc** ┃ blue:**model.conv_units** ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
│ workspace               │ 0.9151 │ 24               │
│ 7317bc6                 │      - │ 16               │
│ ├── e2647ef [exp-ee8a4] │ 0.9145 │ 64               │
│ ├── 15c9451 [exp-a9be6] │ 0.9102 │ 32               │
│ ├── 9c32227 [exp-17dd9] │ 0.9167 │ 256              │
│ ├── 8a9cb15 [exp-29d93] │ 0.9134 │ 128              │
│ ├── dfc536f [exp-a1bd9] │ 0.9151 │ 24               │
│ └── 1a1d858 [exp-6dccf] │ 0.9152 │ 16               │
└─────────────────────────┴────────┴──────────────────┘
```

## 🔏 Persisting experiments

> After selecting an experiment from the table, you can create a Git branch that

Reviewer 2: we should create a sense that it's only one of the possible ways I think

contains the experiment with all its related files.

```dvc
$ dvc exp branch exp-05e87 "cnn-256"
Git branch 'cnn-256' has been created from experiment 'exp-05e87'.
To switch to the new branch run:

        git checkout cnn-256
```

> You can then checkout and continue from working this branch as usual.

Reviewer: Maybe mention that you could also merge it? Again, not important, and I understand it's sort of implied by "working this branch as usual," but it's probably a common workflow.

> You can then checkout and continue from working this branch as usual.

Reviewer: ```suggestion
You can then checkout and continue working from this branch as usual.
```

Author: Thank you. :)

---
title: 'Get Started: Experiments'
---

# Get Started with Experiments

> <abbr>Experiments</abbr> proliferate quickly in ML projects where there are many

Reviewer 2: or data?

Author: This is the Experiments Trail :)

> parameters to tune or other permutations of the code. We can organize such

Reviewer 2: `We can organize ...` - I'm not sure we need this sentence, and `organizing` project usually means using some specific structure (like https://github.com/drivendata/cookiecutter-data-science) l... at least that the way I would read this

Author: Removed that part in 4f56d8f4

projects and keep only what we ultimately need with `dvc experiments`. DVC can

> track experiments for you so there's no need to commit each one to Git. This way

Reviewer 2: New users that come here won't understand this part `to commit each one to Git.`. Committing experiments to Git is very DVC-specific.

Reviewer: Agreed. This is also an issue below when introducing _ephemeral_ experiments. Need to think of how to introduce experiments without assuming user knows about dvc's gitops workflow.

Author: I've updated the intro in 4f56d8.

your repo doesn't become polluted with all of them. You can discard experiments
once they're no longer needed.

> Previously, we learned how to tune [ML pipelines](/doc/start/data-pipelines) and

Reviewer 2: I think one of the main goals of using separate "Trails" (I love the idea and this name!) - is to make them completely independent. I would not distract people by these back-references. We can simplify this.

Author: These two paragraphs are mostly from the older docs. I'll rewrite them. Thank you.

Author: Removed the references in 4f56d8f4

[compare the changes](/doc/start/metrics-parameters-plots). In this section, we

> will explore the basic features of DVC experiment management with

Reviewer 2: can we explain this from the "what users get with this " perspective? E.g. fast local iterations and ability to pick the best? or something like this? otherwise we explain Experiments with experiments management - it's not very informative I'm afraid.

[`get-started-experiments`][gse] project.

[gse]: https://github.com/iterative/get-started-experiments

> ## ⚙️ Installing the Example Project

Reviewer 2: (disclaimer: I haven't participated actively in the discussion so far, so sorry if I repeat certain concerns/arguments)

I think second main objective we should be building these documents with- focus on main things first. In this case it's `dvc exp run` - I should be able to see it in the fist screen with a table. Everything else should come as expandable and/or after sections.

My point here is that get-started are not regular tutorials, they should have a bit different structure. Reproducibility can go in the Katakoda, for example.

Author: There are various aspects of this question:

- Being fully reproducible is better than hiding some commands I think. If someone doesn't know about `dvc pull` (or `tar`, or `git clone`...) and we don't mention here, that may be a blocker for them. ("This thing doesn't work!!") A _veteran_ can simply skip these but a novice may drown in unknown waters. I prefer to err on the side of novices in this section of the docs. 

- I agree with you hiding these in `details` section, and other versions had these hidden. In the discussion, I thought there is _partial consensus_ to move the preparation to the main text by simplifying it. We can discuss this here I believe. @dberenbaum @casperdcl @jorgeorpinel

Author: Updated the span of `details` in 9c324a060665f01de3c4210d9ab86551ab4c8088

These commands are run in the [`get-started-experiments`][gse] project. You can
run the commands in this document after cloning the repository and installing
the requirements.

> ### 👥 Clone the project and create virtual environment

Reviewer 2: emojis: I think we should use the basic style that we already have (less emojis), it's a bit too playful now :)

Author: Too playful, you say :))

In the first version I didn't put any and it looked a bit grim to me. The current emoji set doesn't correspond to these topics. I can put a "gear" here but gear here, gear there, it becomes too gearful. :)

Let me look at this.

Please clone the project and create a virtual environment.

> We create a virtual environment to keep the libraries we use isolated from the
> rest of your system. This prevents version conflicts.

```dvc
$ git clone https://github.com/iterative/get-started-experiments -b pipeline-added
$ cd get-started-experiments
$ virtualenv .venv
$ . .venv/bin/activate
$ python -m pip install -r requirements.txt
```

### 📀 Get the data set

The repository we cloned doesn't contain the dataset. In order to get the
dataset we use `dvc pull` to update the missing data files. `dvc pull` is used
in DVC repositories to update the missing data dependencies.

```dvc
$ dvc pull
```

Then we extract this file that contains labeled images.

```dvc
$ tar -xvzf data/images.tar.gz --directory data/
```

The repository already contains the necessary configuration to run the
experiments. If you would like to learn how to configure a project to run DVC
experiments, please refer to the last section.

<details>

## ⏲ Preparing a project for DVC experiments

In this document we assume that there is already a configured DVC project to

> simplify the introduction. DVC experiments are a feature added in DVC 2.0 and

Author: removed it. thank you.

requires a DVC pipeline defined in the project.

> Please refer to [install](/doc/install) if DVC is not installed in your system.

Author: User may have other preferences about using DVC with venv. I usually use a system-wide DVC. Easier to upgrade and saves taking up space for each project. `get-started-experiments` doesn't list DVC in its `requirements.txt`. If it's the recommended / best-practice to install DVC to `venv`, I should add it to `requirements.txt` too.

> Please refer to [install](/doc/install) if DVC is not installed in your system.

Reviewer: Right. I don't think it's necessarily recommended, just an idea if we want to save a step in setup.

If DVC is not initialized before in the project, you can do so by:

```dvc
$ dvc init
```

DVC also requires commands to be run and their dependencies to be defined as
stages. We use `dvc stage add` to add a stage and set its dependencies.

```dvc
$ dvc stage add -n train \
                -p model.conv_units \
                -p train.epochs \
                -d data/images \
                -m metrics.json \
                python3 src/train.py
```

The command tells DVC to create an experiment named `train`, and that for any
change in `data/images/`, `model.conv_units` or `train.epochs`, we run an
experiment using `src/train.py` that produces a new `metrics.json` file.

Note that the parameters (added with `-p`) are in the default parameters file
`params.yaml` and used in the code as normal, by reading the file. DVC only
tracks the changes and updates them with `--set-param`. For other files that
contain the parameters, DVC expects the filename to be provided similar to

> `-p myparams.yaml:epochs`.

Reviewer: Are `--set-param` or `-p myparams.yaml:epochs` important here?

Author: Removed it. _Preparing your project for DVC_ shouldn't be that detailed. Thank you.

DVC is ready to run the experiments now!

</details>

## 👟 Running the experiment with default parameters

> The purpose of `dvc exp` subcommands is to run the pipeline for ephemeral

Reviewer: Maybe we don't even need _ephemeral_ since it's only really useful to contrast to the `dvc repro` workflow. If we do keep _ephemeral_, we could make it more generic, like "By _ephemeral_ we mean the experiments can be easily discarded."

Author: Removed ephemeral by joining the explanation sentence. :+1:

experiments. By _ephemeral_ we mean the experiments can be run without
committing parameter and dependency changes to Git. Instead the artifacts
produced for each experiment are tracked by DVC and persisted on demand.

Running the experiment with default project settings requires only the command:

```dvc
$ dvc exp run
...
Reproduced experiment(s): exp-7683f
Experiment results have been applied to your workspace.
...
```

It runs the specified command that writes the metrics values to `metrics.json`.

<details>

> ### 📜 If you used `dvc repro` before

Reviewer: _ephemeral_ might make more sense in this section.

Author: It looks it's a sticky word to be avoided :) I kept this the same.

doesn't use any special objects to track the experiments. When you have large
number of experiments that you don't want to commit all to Git, it's better to
use `dvc exp run`. It allows to change the parameters quickly, can track the
history of artifacts and has facilities to compare these experiments easily.

`dvc repro` is still available to run the pipeline when these extra features are
not needed.

</details>

## 🧥 Running the experiment by setting parameters

Now let's do some more experimentation.

DVC allows to update the parameters defined in the pipeline without modifying
the files manually. We use this feature to set the convolutional units in
`train.py`.

```dvc
$ dvc exp run --set-param model.conv_units=24
...
Reproduced experiment(s): exp-6c06d
Experiment results have been applied to your workspace.

To promote an experiment to a Git branch run:

        dvc exp branch <exp>
```

When you run `dvc exp run` with `--set-param`, it updates the parameter file. We
can see the effect of it by looking at the diff.

```dvc
$ git diff params.yaml
```

```git
-model:
-  conv_units: 16
+model:
+  conv_units: 24

> ```

Reviewer: This is a nice explanation, but could it be hidden?

> ## 🏃‍♂️🏃🏾‍♂️🏃🏻‍♂️ Run multiple experiments in parallel

Reviewer 2: Can we merge three different "running experiments sections" into one or at least to? Otherwise it becomes more of a user guide I think. No intention here to be very comprehensive, we need to make it clear what is possible, ideally with one single command, one single table

Author: I rather would like to keep the "sales pitch" in these docs, and IMO that's "running experiments in parallel" for `dvc exp run`. But in order to tell "you can run experiments in parallel", we first need to tell "you can run experiments".

I think keeping these in separate sections emphasize _multiple experiments_. If I merge these, `--queue` option will become a minor thing in a deeper paragraph.

Reviewer: I think keeping the "sales pitch" could mean having a compelling header of a hidden section or otherwise calling out the idea, but keeping how to do it in the hidden section. You can still have all these experiments in the `dvc exp show` table even if they aren't in the main text. Having that table higher up seems important since it's unclear how `dvc exp run` is helpful until that point.

Author: I put the sections in `details`. Let's see how it looks.

$ dvc exp run --queue -S model.conv_units=128
Queued experiment 'ac66940' for future execution.
$ dvc exp run --queue -S model.conv_units=256
Queued experiment '8bb6049' for future execution.
```

Next, run all (`--run-all`) queued experiments in parallel. You can specify the
number of parallel processes using `--jobs`:

```dvc
$ dvc exp run --run-all --jobs 2
```

## ↔️ Comparing experiments

The experiments are run several times with different parameters. We use
`dvc exp show` to compare all of these experiments. This command presents the
parameters and metrics produced in experiments in a nicely formatted table.

```dvc
$ dvc exp show
```

![](/img/start-dvc-exp-show-210704.png)

By default it shows all the parameters and the metrics with the timestamp. If
you have large number of parameters, metrics or experiments, this may lead to a
cluttered view. You can limit the table to specific metrics, or parameters, or
hide the timestamp column with `--include-metrics`, `--include-params`, or
`--no-timestamp` options of the command, respectively.

```dvc
$ dvc exp show --no-timestamp \
  --include-params model.conv_units --include-metrics acc
```

> ![](/img/start-dvc-exp-show-no-timestamp-210704.png)

Reviewer: This is another part that might be a candidate for a hidden section.

## 🔏 Persisting experiments

After selecting an experiment from the table, you can commit the hyperparameters
and other dependencies that produced this successful experiment to your Git
history.

`dvc exp apply` brings back all specific artifacts and parameters from the
experiment to the <abbr>workspace</abbr>.

```dvc
$ dvc exp apply exp-cb13f
Changes for experiment 'exp-cb13f' have been applied to your current workspace.
```

We can see the changes in the repository and commit them to Git.

```dvc
$ git diff
$ git add .
$ git commit -m "Successful experiment"
```

> It may also be desirable to commit a particular experiment to a Git branch

Reviewer: Do we need to introduce both `apply` and `branch` here? It seems sufficient to pick one as a default and save the other for the UG.

Author: removed `dvc exp apply` and kept `branch`, as it's the option advertised in the command output as well.

directly, without bringing to the workspace.

```dvc
$ dvc exp branch exp-05e87 "cnn-256"
Git branch 'cnn-256' has been created from experiment 'exp-05e87'.
To switch to the new branch run:

        git checkout cnn-256
```

You can then checkout and continue from working this branch as usual.

> ## 📛 Note on experiment names

Reviewer 2: It think this it a bit too much - to have the whole section like this as a Note. Is there a very important reason to make it of the same importance as "running experiments" and other sections?

Can we hide it?

Author: The reason I put this into a separate section is @aguschin's comments about experiments names, I think in iterative/dvc#6050 . For a newcomer having two names, one reported by `--queue`, one given after the run is a bit confusing and I'd rather have a single autogenerated commit name for both of these. Current behavior is a bit confusing for the newcomer. 

I can move this to a relevant UG page and hide here.

Reviewer: This is in the current dvc sprint to fix, and I just updated the importance. Let's assume this will be fixed before publication, and we can bump the priority further if needed. Hopefully, we can hide or drop this whole section or at least greatly simplify.

Reviewer: cc @karajan1001. If you have capacity, please pick up this issue next.

Author: removed the naming section

When you create an experiment, DVC generates a hash value from the contents of
the experiment. This is shown when you use `--queue` option, e.g.,

```dvc
$ dvc exp run --queue -S model.conv_units=32
Queued experiment '6518f17' for future execution.
```

After _running_ the experiment, DVC uses another auto-generated name to refer to
the experiment. Typically these start with `exp-`, and can be set via
`--name / -n` option of `dvc exp run`. So when you add an experiment by setting
the name, you can see the hash value as _queued experiment_:

```dvc
$ dvc exp run --queue --name cnn-512 -S model.conv_units=512
Queued experiment '86bd8f9' for future execution.
```

In `dvc exp show` you can see both of these names:

```dvc

> $ dvc exp show --no-pager --no-timestamp \

Reviewer 2: we need to fix the highlighter? cc @jorgeorpinel

Reviewer 6: Why? What's wrong? Sorry, review app isn't up rn.

      --include-metrics loss --include-params model.conv_units

┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
┃ Experiment              ┃ loss    ┃ model.conv_units ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
│ workspace               │ 0.23534 │ 64               │
│ 3973b6b                 │ -       │ 16               │
│ ├── aeaabb0 [exp-cb13f] │ 0.23534 │ 64               │
│ ├── d0ee7ce [exp-5dccf] │ 0.23818 │ 32               │
│ ├── 1533e4d [exp-88874] │ 0.24039 │ 128              │
│ ├── b1f41d3 [cnn-256]   │ 0.23296 │ 256              │
│ ├── 07e927f [exp-6c06d] │ 0.23279 │ 24               │
│ ├── b2b8586 [exp-2a1d5] │ 0.25036 │ 16               │
│ └── *86bd8f9            │ -       │ 512              │
└─────────────────────────┴─────────┴──────────────────┘
```

> When an experiment is not run yet, only the former hash value is shown.

Reviewer: To add to the confusion 🤣 , the hash value changes before and after the experiment is run. I don't think it's worth getting into any of these details in this doc.
